### PR TITLE
planner: clean BindRecord in the binding package

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #48875

Problem Summary: planner: clean BindRecord in the binding package

### What changed and how does it work?

NO LOGICAL CHANGE, JUST REFACTOR

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/bindinfo/global_handle.go [hunk 1]
```diff
@@ -443,54 +429,38 @@ func lockBindInfoTable(sctx sessionctx.Context) error {
...
 
-// tmpBindRecordMap is used to temporarily save bind record changes.
-// Those changes will be flushed into store periodically.
-type tmpBindRecordMap struct {
-	sync.Mutex
-	atomic.Value
-	flushFunc func(binding Binding) error
+// invalidBindingCache is used to store invalid bindings temporarily.
+type invalidBindingCache struct {
+	mu sync.Mutex
```
Comment (hawkingrei) [inline pkg/bindinfo/global_handle.go:434]:
```suggestion
	mu sync.RWMutex
```

File: pkg/bindinfo/global_handle.go [hunk 2]
```diff
@@ -122,9 +121,7 @@ type globalBindingHandle struct {
 	// This value is used to avoid reload duplicated bindings from storage.
 	lastUpdateTime atomic.Value
```
Comment (elsa0520) [inline pkg/bindinfo/global_handle.go:123]: Keep the comments ~

**Review by elsa0520 — APPROVED** [review]
LGTM

**Pull request merged**